src/app/rmmapi/rmmauthguard.service.ts

```diff
@@ -58,18 +60,34 @@ export class RMMAuthGuardService implements CanActivate, CanActivateChild {
         });
     }
 
-    isLoggedIn(): Observable<boolean> {
+    isLoggedIn(): Observable<boolean | UrlTree> {
         return this.http.get('/rest/v1/me').pipe(
             map((res: any) => {
                 this.wasLoggedIn = res && res.status === 'success';
-                return this.wasLoggedIn;
+                if (res && res.result) {
+                    this.currentMe = res.result;
+                }
+                return this.checkStatus();
             }),
         );
     }
 
-    canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): boolean | Observable<boolean> | Promise<boolean> {
+    checkStatus(): boolean | UrlTree {
+        // Expired users are only allowed to visit account/subscriptions
+        if (this.currentMe && this.currentState
+            && this.currentMe.account_status === 'expired'
```

Comment:
Is it possible to do `this.currentMe.isExpired()` instead?

Reply:
Nope, its not a RunboxMe obj, just a chunk of data (and it seems overkill to create the object to me)